node/src/main/scala/org/bitcoins/node/networking/P2PClient.scala

```diff
@@ -90,16 +100,20 @@ case class P2PClientActor(
       //after receiving Tcp.Connected we switch to the
       //'awaitNetworkRequest' context. This is the main
       //execution loop for the Client actor
-      val _ = handleCommand(cmd, peer = None)
+      handleCommand(cmd, peer = None)
 
     case connected: Tcp.Connected =>
-      val _ = handleEvent(connected, unalignedBytes = ByteVector.empty)
+      Await.result(handleEvent(connected, unalignedBytes = ByteVector.empty),
```

Comment:
Same as above

Reply:
So this `Await` is here on purpose. Actor mailboxes messages are queued and processing synchronously. If we return a `Future`, it looks like we have completed processing. This `Await` forces the future to be completed before we return, so the Actor doesn't move onto the next message

Reply:
But won't this cause everything else to stop? I'm assuming Akka supports a pattern like this without blocking threads

Reply:
I can confirm these are needed, and haven't found a alternative yet

Reply:
https://stackoverflow.com/questions/36592074/wait-on-asynchronous-future-call-before-processing-next-message-in-akka